libraries/stdlib/src/kotlin/collections/AbstractIterator.kt

```diff
@@ -13,8 +13,8 @@ private enum class State {
 }
 
 /**
- * A base class to simplify implementing iterators so that implementations only have to implement [[computeNext()]]
- * to implement the iterator, calling [[done()]] when the iteration is complete.
+ * A base class to simplify implementing iterators so that implementations only have to implement [computeNext]
```

Comment:
Warning, using this class has performance penalty compared to plain raw implementation per case, measured with JMH when operation per element is relatively cheap.